### PR TITLE
[frontend] Add an API route to the cloud upload controller

### PR DESCRIPTION
**Pull request by bgeuken**
This adds API routes (aka xml responses) for:

  * Creation of upload jobs
  * Listing all upload jobs a user has access to

We are re-using the existing webui controller, unlike we usually do, to
reduce code duplication.

### CODE REVIEW

**Comment by bgeuken** [comment]
[P13: Cloud upload API](https://trello.com/c/iZ4Q1X7v/712-p13-cloud-upload-api)

**Comment by bgeuken** [comment]
Blocked for now. I have some trouble testing the new API routes on a real machine

File: src/api/app/models/cloud/params/ec2.rb [hunk 1]
```diff
@@ -15,6 +17,14 @@ def self.build(params)
         new(params.slice(:region, :ami_name))
       end
 
+      def to_xml(options = {})
```
Comment (ChrisBr) [inline src/api/app/models/cloud/params/ec2.rb]: Why is this necessary? I'm missing where it is used?

File: src/api/app/models/cloud/upload_job.rb [hunk 2]
```diff
@@ -39,6 +41,16 @@ def target_validator_class
       @target_validator_class ||= "::Cloud::Params::#{target.capitalize}".constantize
     end
 
+    def to_xml(options = {})
```
Comment (ChrisBr) [inline src/api/app/models/cloud/upload_job.rb]: Why is this necessary? I'm missing where it is used?

File: src/api/app/models/cloud/params/ec2.rb (hunk 1 above)
Comment (bgeuken) [reply src/api/app/models/cloud/params/ec2.rb]: This is used in the [to_xml method of the UploadJob model](https://github.com/openSUSE/open-build-service/pull/4691/files#diff-53550eae0633ce2e45f0e84869c7fb86R50)

File: src/api/app/models/cloud/upload_job.rb (hunk 2 above)
Comment (bgeuken) [reply src/api/app/models/cloud/upload_job.rb]: And this is used when rendering the upload jobs in [the controller](https://github.com/openSUSE/open-build-service/pull/4691/files#diff-4016cd1c604738f10a2f27ccf98bea94R34)

**Comment by bgeuken** [comment]
I created a distinct api controller for now, since I didn't find time to refactor, and test, the webui authentication code
Moved to -> #4739
